skaffold-plugins-core/src/main/java/com/google/cloud/tools/skaffold/yaml/SkaffoldYamlGenerator.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2018 Google LLC. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.skaffold.yaml;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import java.nio.file.Path;
+
+/** Automatically generates contents of a skaffold.yaml. */
+public class SkaffoldYamlGenerator {
+
+  private ImmutableList<Path> manifestPaths;
+
+  /**
+   * Creates a new {@link SkaffoldYamlGenerator}.
+   *
+   * @param manifestPaths a non-empty list of paths to kubernetes yamls. Supports glob pattern
```

Comment:
Hmm, the glob matching seems an odd choice.  I'd rather provide a helper to expand a glob to a set of files.  Besides the code below doesn't seem to support glob patterns!

Reply:
Really it's skaffold that supports glob matching. I guess I should just remove the comment?

Reply:
It might make more sense to have the `manifestPaths` be of type `String` rather than `Path`.